burn-import/src/burn/graph.rs

```diff
@@ -472,17 +583,34 @@ impl<PS: PrecisionSettings> BurnGraph<PS> {
     }
 }
 
-#[derive(new)]
+#[derive(new, Debug)]
 struct BurnGraphState<'a, PS: PrecisionSettings> {
     nodes: &'a Vec<Node<PS>>,
 }
 
-impl<'a, PS: PrecisionSettings> Serialize for BurnGraphState<'a, PS> {
+/// For custom serialization of the graph state in module struct, there are generally
+/// two options:
+/// 1. Serialize the nodes as a map with the node name as the key and the node as the value.
+/// 2. Serialize the nodes as a tuple.
+///
+/// PrettyJson, NamedMpk, and NamedMpkGz use the first option and this struct is used for that.
+struct StructMap<'a, PS: PrecisionSettings>(BurnGraphState<'a, PS>);
+
+/// For custom serialization of the graph state in module struct, there are generally
+/// two options:
+/// 1. Serialize the nodes as a map with the node name as the key and the node as the value.
+/// 2. Serialize the nodes as a tuple.
+///
+/// Mpk, and Bincode use the second option and this struct is used for that.
```

Comment:
We should explain the reason why those recorders use those options.

Reply:
OK. I'll rewrite my doc comments.